### DISCUSSION TITLE
posts/2023-04-08-nextjs-appdir-blog

### DISCUSSION THREAD

**Discussion by giscus** [cmdcolin/cmdcolin.github.io#14] [General]
# posts/2023-04-08-nextjs-appdir-blog

A blog

https://cmdcolin.github.io/posts/2023-04-08-nextjs-appdir-blog

**Comment by yashkc2025** [comment]
Hi Colin,

This is a great tutorial. But, if I am trying to declare a component client side. Next is throwing the error 'unable to resolve module : fs'.

Even if the client side component isn't using fs or api.ts.

How can i fix it

**Reply by cmdcolin** [reply]
thanks for the kind words. I think it is tricky to juggle client and server components, this blog talks about some practical considerations https://www.mux.com/blog/what-are-react-server-components

if you have a minimal reproduction i can check it out but just remember that any child of a client component is a client component unless you structure it in a particular way...i'm still wrapping my head around the way that server component can be a child of client component

**Reply by yashkc2025** [reply]
Thank you for replying, Colin. I checked the article you provided and it was awesome. 

```
I'm still wrapping my head around the way that server component can be a child of client component
```
Yup, 😅 I am also learning new things like this everyday, but thanks for the tutorial.
For now, I have went ahead with a different approach.

Bye,
Yash

**Comment by robowager** [comment]
Thank you for the great tutorial.
